graphql-cli/src/test/java/io/ballerina/graphql/cmd/ServiceGenerationTest.java

```diff
@@ -0,0 +1,201 @@
+            file.setReadable(false);
+            String[] args = {"-i", graphqlSchema.toString(), "-o", tmpDir.toString(), "-m", "service"};
+            GraphqlCmd graphqlCmd = new GraphqlCmd(printStream, tmpDir, false);
+            new CommandLine(graphqlCmd).parseArgs(args);
+            String message = String.format(MESSAGE_CAN_NOT_READ_SCHEMA_FILE, graphqlSchema);
+            graphqlCmd.execute();
+            String output = readOutput(true);
+            Assert.assertTrue(output.contains(message));
+        } catch (BLauncherException | IOException e) {
+            Assert.fail(e.toString());
+        }
+    }
+
+    @DataProvider(name = "schemaFiles")
+    public Object[] createSchemaFilesData() {
+        return new Object[]{"SchemaWithBasic01Api.graphql", "SchemaWithBasic02Api.graphql",
+                "SchemaWithInputsApi.graphql", "SchemaWithMutationApi.graphql", "SchemaWithSubscriptionApi.graphql",
+                "SchemaWithBasic03Api.graphql", "SchemaWithEnumApi.graphql", "SchemaWithUnionApi.graphql",
+                "SchemaWithInterfaceApi.graphql", "SchemaWithMultipleInterfacesApi.graphql",
+                "SchemaWithInterfacesImplementingInterfacesApi.graphql", "SchemaWithMultiDimensionalListsApi.graphql",
+                "SchemaWithDefaultParameters01Api.graphql", "SchemaWithDefaultParameters02Api.graphql",
+                "SchemaWithDefaultParameters03Api.graphql", "SchemaWithDefaultParameters04Api.graphql",
+                "SchemaDocsWithQueryResolversApi.graphql",
+                "SchemaDocsWithMutationAndSubscriptionResolversApi.graphql",
+                "SchemaDocsWithResolverMultipleLinesApi.graphql", "SchemaDocsWithResolverArgumentsApi.graphql",
+                "SchemaDocsWithMultipleLinesApi.graphql", "SchemaDocsWithOutputsApi.graphql",
+                "SchemaDocsWithUnionApi.graphql", "SchemaDocsWithEnumApi.graphql", "SchemaDocsWithInputsApi.graphql",
+                "SchemaDocsWithInterfacesApi.graphql", "SchemaDocsWithDeprecated01Api.graphql",
+                "SchemaCompleteApi.graphql"};
+    }
+
+    @Test(description = "Test compilation for all schemas, method - default", dataProvider = "schemaFiles")
+    public void testCompilationForAllSchemas(String file) {
+        Path schemaPath = this.resourceDir.resolve(Paths.get("serviceGen", "graphqlSchemas", "valid", file));
+
+        String[] args = {"-i", schemaPath.toString(), "-o", this.tmpDir.toString(), "--mode", "service"};
+        GraphqlCmd graphqlCmd = new GraphqlCmd(printStream, this.tmpDir, false);
+        new CommandLine(graphqlCmd).parseArgs(args);
+
+        try {
+            graphqlCmd.execute();
+            DiagnosticResult diagnosticResult = getDiagnosticResult(this.tmpDir);
+            Assert.assertTrue(hasOnlyFuncMustReturnResultErrors(diagnosticResult.errors()));
+        } catch (BLauncherException e) {
+            String output = e.toString();
+            Assert.fail(output);
```

Comment:
There are some inconsistencies in checking the exception and returning/asserting the error. Shall we check those and make them consistent?

Reply:
fixed it.